File: src/main/java/info/dddeurope/lab/Repositories/SocialGraph.java
package info.dddeurope.lab.Repositories;

import org.jgrapht.*;
import org.jgrapht.alg.interfaces.ShortestPathAlgorithm.SingleSourcePaths;
import org.jgrapht.alg.shortestpath.DijkstraShortestPath;
import org.jgrapht.graph.*;

public class SocialGraph {

    Graph<String, DefaultEdge> graph; 

    public SocialGraph() {
        this.graph = new DefaultDirectedGraph<>(DefaultEdge.class);
    }
    
    public void addPerson(String id) {
        this.graph.addVertex(id);
    }

    public void createFriendship(String personIdA, String personIdB) {
        this.graph.addEdge(personIdA, personIdB);
        this.graph.addEdge(personIdB, personIdA);
    }

    public void breakFriendship(String personIdA, String personIdB) {
        this.graph.removeEdge(personIdA, personIdB);
        this.graph.removeEdge(personIdB, personIdA);
    }

    public GraphPath<String,DefaultEdge> getConnections(String userA, String userB) {
        DijkstraShortestPath<String, DefaultEdge> dijkstraAlg =
            new DijkstraShortestPath<>(this.graph);
        SingleSourcePaths<String, DefaultEdge> iPaths = dijkstraAlg.getPaths(userA);
        return iPaths.getPath(userB);
    }

}

File: DotNet/App/Views/SocialGraph.cs
using System.Collections.Generic;
using QuickGraph;
using QuickGraph.Algorithms;

namespace App.Views
{
    public class SocialGraph
    {
        readonly BidirectionalGraph<string, Edge<string>> _graph;

        public SocialGraph()
        {
            _graph = new BidirectionalGraph<string, Edge<string>>();
        }

        public void AddPerson(string id)
        {
            _graph.AddVertex(id);
        }

        public void CreateFriendship(string personIdA, string personIdB)
        {
            _graph.AddEdge(new Edge<string>(personIdA, personIdB));
            _graph.AddEdge(new Edge<string>(personIdB, personIdA));
        }

        public void BreakFriendship(string personIdA, string personIdB)
        {
            _graph.RemoveEdge(new Edge<string>(personIdA, personIdB));
            _graph.RemoveEdge(new Edge<string>(personIdB, personIdA));
        }

        public IEnumerable<Edge<string>> GetConnections(string userA, string userB)
        {
            var paths = _graph.ShortestPathsDijkstra(edge => 1, userA);
            paths(userB, out var result);
            return result;
        }
    }
}